sdks/python/apache_beam/ml/inference/pytorch.py

```diff
@@ -37,7 +37,7 @@ def __init__(self, device: torch.device):
     self._device = device
 
   def run_inference(self, batch: List[torch.Tensor],
-                    model: torch.nn.Module) -> Iterable[torch.Tensor]:
+                    model: torch.nn.Module) -> Iterable[PredictionResult]:
     """
```

Comment:
If the contract is that this should produce an iterable, I wonder if we should avoid materializing the list by returning a generator?

Reply:
I think its done in the later stage in a DoFn 

https://github.com/apache/beam/blob/68b9100db4b82196eba025bd417c46d03656f7c0/sdks/python/apache_beam/ml/inference/base.py#L210

Reply:
What I'm suggesting is that we could avoid materializing a list between here and there by returning a generator (`(PredictionResult(x, y) for x, y in zip(batch, predictions))`) here. It looks like that would be undermined by the base DoFn materializing a list again here anyway: https://github.com/apache/beam/blob/68b9100db4b82196eba025bd417c46d03656f7c0/sdks/python/apache_beam/ml/inference/base.py#L198

Reply:
Ah got it.